### PR TITLE
Build/dsp in arm

### PR DESCRIPTION
**Pull request by m-m-adams**
Ready to go - builds DSP code in ARM, using a unity build in place of LTO. Significant bin size reduction, down to 1.1mb

+10 voices compared to pure thumb, returning to the arm performance level

### CODE REVIEW

File: src/deluge/dsp/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    NE10
+    eyalroz_printf
+)
+target_compile_options(deluge_dsp PRIVATE
+    ${DELUGE_COMMON_COMPILE_OPTIONS}
+    # use a unity build instead (combine all cpp to one before compiling), which has similar gains without the arm/thumb issues
+    "-fno-lto"
+    # we're not using lto so go ahead and inline
+    "-finline-functions"
+    # Build dsp code with thumb mode
```
Comment (sapphire-arches) [inline src/deluge/dsp/CMakeLists.txt:39]:
```suggestion
    # Build dsp code with arm mode
```

File: src/deluge/dsp/delay/delay.cpp [hunk 2]
```diff
@@ -32,17 +29,8 @@ Delay::Delay() {
...
-	if (!song) {
-		song = currentSong;
-	}
-	if (song) {
-		syncLevel = (SyncLevel)(8 - (song->insideWorldTickMagnitude + song->insideWorldTickMagnitudeOffsetFromBPM));
-	}
-	else {
-		syncLevel = (SyncLevel)(8 - FlashStorage::defaultMagnitude);
-	}
+	syncLevel = (SyncLevel)5;
```
Comment (sapphire-arches) [inline src/deluge/dsp/delay/delay.cpp:32]: Does this get overwritten later?

File: src/deluge/CMakeLists.txt [hunk 3]
```diff
@@ -7,6 +8,9 @@ file(GLOB_RECURSE deluge_SOURCES CONFIGURE_DEPENDS
   ${CMAKE_CURRENT_LIST_DIR}/*.cpp
 )
 
+# Filter out GUI sources as those are built as part of the deluge_gui library
```
Comment (sapphire-arches) [inline src/deluge/CMakeLists.txt:11]:
```suggestion
# Filter out DSP sources as those are built as part of the deluge_dsp library
```

File: src/deluge/dsp/delay/delay.cpp (hunk 2 above)
Comment (m-m-adams) [reply src/deluge/dsp/delay/delay.cpp:32]: Yeah it's set by mod controllable audio/global effectable now in order to strip out all the dependencies on song and param

**Pull request merged**